BUG Sphinx `docname` from source-read event is sometimes `None`

Doc build and several `test_full.py` tests on Windows CI workflow failing on recent PRs (e.g., #1173, #1172)

Windows `test_full.py` failures are all due to:
```
sphinx_gallery\gen_gallery.py:919: in write_api_entry_usage
    "sg_api_usage" not in docname
E   TypeError: argument of type 'NoneType' is not iterable
```
Similarly doc build error is:
```
An error happened in rendering the page advanced.
Reason: AttributeError("'TocTree' object has no attribute 'get_toctree_ancestors'")
```
AFAICT `docname` should: "Returns the docname of the document currently being parsed."
https://www.sphinx-doc.org/en/master/extdev/envapi.html#sphinx.environment.BuildEnvironment.docname

'source-read' event emits `env.docname` which  does not seem to have changed recently: https://github.com/sphinx-doc/sphinx/blob/da9f8a5c33ad5eeef05dd780f2988f3ff7351ef3/sphinx/environment/__init__.py#L545

Seems to be a Sphinx issue as there was a release some hours ago.

--- a/sphinx_gallery/docs_resolv.py
+++ b/sphinx_gallery/docs_resolv.py
@@ -85,6 +85,8 @@ def parse_sphinx_docopts(index):
 
     pos = index.find("var DOCUMENTATION_OPTIONS")
     if pos < 0:
+        pos = index.find("const DOCUMENTATION_OPTIONS")  # Sphinx 7.2+
+    if pos < 0:
         raise ExtensionError("Documentation options could not be found in index.")
     pos = index.find("{", pos)
     if pos < 0:

--- a/sphinx_gallery/gen_gallery.py
+++ b/sphinx_gallery/gen_gallery.py
@@ -909,6 +909,7 @@ def write_api_entry_usage(app, docname, source):
     connects all of the API entries in a module to the examples
     that they are used in.
     """
+    docname = docname or ""  # can be None on Sphinx 7.2
     gallery_conf = app.config.sphinx_gallery_conf
     if gallery_conf["show_api_usage"] is False:
         return

